suse2021-ns/fo/xref.xsl

```diff
@@ -158,58 +158,74 @@
   </fo:inline>
 </xsl:template>
 
+
+<xsl:template match="node()" mode="copy-fo">
+  <xsl:copy>
+    <xsl:apply-templates mode="copy-fo"/>
+  </xsl:copy>
+</xsl:template>
+
+<xsl:template match="@*" mode="copy-fo"/>
+
 <xsl:template match="d:xref" name="xref">
-  <xsl:variable name="targets" select="key('id',@linkend)"/>
-  <xsl:variable name="target" select="$targets[1]"/>
-  <xsl:variable name="refelem" select="local-name($target)"/>
-  <xsl:variable name="target.book" select="($target/ancestor-or-self::d:article|$target/ancestor-or-self::d:book)[1]"/>
-  <xsl:variable name="this.book" select="(ancestor-or-self::d:article|ancestor-or-self::d:book)[1]"/>
-  <xsl:variable name="lang" select="(ancestor-or-self::*/@lang|ancestor-or-self::*/@xml:lang)[1]"/>
+  <xsl:variable name="context" select="."/>
+  <xsl:variable name="target" select="key('id', @linkend)[1]"/>
   <xsl:variable name="xref.in.samebook">
-    <xsl:call-template name="is.xref.in.samebook">
+    <xsl:call-template name="is.xref.within.rootid">
       <xsl:with-param name="target" select="$target"/>
     </xsl:call-template>
   </xsl:variable>
 
+  <xsl:variable name="lang" select="(ancestor-or-self::*/@lang|ancestor-or-self::*/@xml:lang)[1]"/>
```

Comment:
Since we're targeting DB5 only, I think you can simplify to this:

```suggestion
  <xsl:variable name="lang" select="(ancestor-or-self::*/@xml:lang)[1]"/>
```